pkg/sql/conn_executor_exec.go

```diff
@@ -298,7 +299,15 @@ func (ex *connExecutor) execStmtInOpenState(
 			typeHints[types.PlaceholderIdx(i)] = coltypes.CastTargetToDatumType(t)
 		}
 		if _, err := ex.addPreparedStmt(
-			ctx, name, Statement{SQL: s.Statement.String(), AST: s.Statement}, typeHints,
+			ctx, name,
+			Statement{
+				Statement: parser.Statement{
+					SQL:             s.Statement.String(),
```

Comment:
I thought you had enabled the storing of the original client-supplied string. Is there a way to use that here instead of re-constructing a string from the AST?